### DISCUSSION TITLE
Enhance postUpgradeTasks to create MR

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [renovatebot/renovate#28883] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab - 37.347.0

### Please tell us more about your question or problem

I am trying to leverage the amazing Renovate to raise a custom change by running a script as part of `postUpgradeTasks`. I can get all to work from the script, but as no changes are identified the `postUpgradeTasks` is not triggered.

One idea is to create a custom data source and make it trigger by doing some wonky stuff like using pipeline variables or something like that so every Renovate execution is a new value, but perhaps there is a better way without the need to have something funky like that.

If you have any ideas, that would be great!

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
The only way I can think of is if you find a way to achieve this with the custom/regex manager

**Reply by juancarlosjr97** [reply]
Any idea how to make a regexManager to always run?

**Reply by rarkins** [reply]
First, it needs to think it found a valid dependency on disk. Then it needs to look up a datasource and find there's an upgrade. Then it needs to think it successfully upgraded/updated. And after that it will call any postUpgradeTasks. You might be able to have a dummy value, let Renovate update it, then reset the update using postUpgrade tasks plus do whatever else you want.
